controllers/configuration_controller.go

```diff
@@ -337,112 +276,148 @@ func (r *ConfigurationReconciler) reconcile(ctx context.Context, cv *v1alpha1.Co
-				NamespacedName: types.NamespacedName{
-					Namespace: d.Namespace,
-					Name:      d.Name,
-				},
-			})
+		ref = &v1alpha1.ComponentVersion{}
+		if err := r.Get(ctx, obj.GetObjectKey(), ref); err != nil {
+			return false
 		}
 
-		return reqs
+	default:
+		// if the APIVersion is not set then default to "delivery.ocm.software/v1alpha1"
+		if obj.APIVersion == "" {
+			obj.APIVersion = v1alpha1.GroupVersion.String()
+		}
+		// if the Namespace is not set then default to the parent object's namespace
+		if obj.Namespace == "" {
+			obj.Namespace = ns
+		}
+		// the dynamic client needs to know the GroupVersionResource for the object it's trying to fetch
+		// so construct that and fetch the unstructured object
+		gvr := obj.GetGVR()
+		src, err := r.DynamicClient.Resource(gvr).Namespace(obj.Namespace).Get(ctx, obj.Name, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		snapshotName, ok, err := unstructured.NestedString(src.Object, "status", "snapshotName")
+		if err != nil {
+			return false
+		}
+		if !ok {
+			return false
+		}
+		// finally get the snapshot itself
+		ref = &v1alpha1.Snapshot{}
+		if err := r.Get(ctx, types.NamespacedName{Namespace: obj.Namespace, Name: snapshotName}, ref); err != nil {
+			return false
+		}
 	}
+	return conditions.IsReady(ref)
 }
 
-func (r *ConfigurationReconciler) indexBy(kind, field string) func(o client.Object) []string {
-	return func(o client.Object) []string {
-		l, ok := o.(*v1alpha1.Configuration)
-		if !ok {
-			panic(fmt.Sprintf("Expected a Localization, got %T", o))
+func (r *ConfigurationReconciler) checkFluxSourceReadiness(ctx context.Context, obj meta.NamespacedObjectKindReference) bool {
+	var ref conditions.Getter
+	switch obj.Kind {
+	case sourcev1.GitRepositoryKind:
+		ref = &sourcev1.GitRepository{}
+		if err := r.Client.Get(ctx, client.ObjectKey{Namespace: obj.Namespace, Name: obj.Name}, ref); err != nil {
+			return false
 		}
+	default:
+		return false
+	}
+	return conditions.IsReady(ref)
+}
 
-		switch field {
-		case "SourceRef":
-			if l.Spec.Source.SourceRef != nil && l.Spec.Source.SourceRef.Kind == kind {
-				namespace := l.GetNamespace()
-				if l.Spec.Source.SourceRef.Namespace != "" {
-					namespace = l.Spec.Source.SourceRef.Namespace
-				}
-				return []string{fmt.Sprintf("%s/%s", namespace, l.Spec.Source.SourceRef.Name)}
-			}
-			return []string{fmt.Sprintf("%s/%s", l.Spec.ComponentVersionRef.Namespace, l.Spec.ComponentVersionRef.Name)}
-		case "ConfigRef":
-			namespace := l.GetNamespace()
-			if l.Spec.ComponentVersionRef.Namespace != "" {
-				namespace = l.Spec.ComponentVersionRef.Namespace
+func makeRequestsForConfigurations(ll ...v1alpha1.Configuration) []reconcile.Request {
+	slices.SortFunc(ll, func(a, b v1alpha1.Configuration) bool {
+		aKey := fmt.Sprintf("%s/%s", a.GetNamespace(), a.GetName())
+		bKey := fmt.Sprintf("%s/%s", b.GetNamespace(), b.GetName())
+		result := []string{aKey, bKey}
+		slices.Sort(result)
+		return aKey == result[0]
```

Comment:
```suggestion
		aKey := fmt.Sprintf("%s/%s", a.GetNamespace(), a.GetName())
		bKey := fmt.Sprintf("%s/%s", b.GetNamespace(), b.GetName())
		return aKey < bKey
```

You can compare strings.